### DISCUSSION TITLE
Opencart-3 main repository

### DISCUSSION THREAD

**Discussion by TheCartpenter** [opencart/opencart#11110] [Ideas]
Should we consider replacing the old recurring to the new subscription built-in? For that, I assume that JS library would also need to be upgraded.

**Comment by TheCartpenter** [comment]
GDPR also while people arises the wait for the next OC v4 version, regulations needs to be added for store owners that needs to deal with their customers' demands.

**Comment by TheCartpenter** [comment]
GDPR information and action fields have now been integrated.

**Comment by TheCartpenter** [comment]
Session expire options have now been integrated into the admin store and setting page.

**Comment by TheCartpenter** [comment]
There seem to be persisting issues going on with custom fields. Will be working on that.

**Comment by TheCartpenter** [comment]
Custom fields bugfixes completed.

**Reply by totalsportUA** [reply]
https://totalsport.ua/

**Comment by TheCartpenter** [comment]
Address format added to the list.

**Comment by TheCartpenter** [comment]
Address format integration completed.

**Comment by TheCartpenter** [comment]
May need to work on the migration of subscriptions.

**Comment by TheCartpenter** [comment]
Integrated currency cron loaders (@condor2 : https://github.com/opencart/opencart-3/commit/3b04cbfdfc50640cf5a83b556158547ec780c457) .

**Comment by TheCartpenter** [comment]
The Event Engine may have to be put up-to-date. Tested the events with PHP v8.1 release with the OC 3 Engine Files and causes ```array|string``` deprecated errors event though ```(string)``` is being used. Tested the difference with the ```__isset``` magic method in the startup, it may have made a difference back then when that was added. It doesn't anymore. Therefore, an upgrade of the Engine will be taken in effect prior to the PHP v7.4 deprecation on November 22, 2022.

**Comment by TheCartpenter** [comment]
@danielkerr: There seem to be a deprecation with the backward API compatibility method you are using in OC-3:

```Unknown: Automatic conversion of false to array is deprecated in admin/controller/sale/order.php on line 388```

Since PHP 8.1 just by loading the admin > sales > orders page. Did a research for this error and, so far, no identified solutions: https://stackoverflow.com/a/71035543

**Comment by TheCartpenter** [comment]
Most PHP 8 compatibility codes have now been integrated into the core. Fixed the language, currency and session libraries. Users can now refresh the page after clearing their browsing data without error messages once the store is reloaded.

**Comment by TheCartpenter** [comment]
Will be working on the design/layout's new extension arrays.

**Comment by TheCartpenter** [comment]
@danielkerr: On August 10, I will be migrating my equipment to a better system to keep working on the repository.

**Comment by TheCartpenter** [comment]
@danielkerr: Migration is complete. This weekend, I should resume the course with the repository.

**Comment by TheCartpenter** [comment]
GDPR integration is complete; exclusing the mail/gdpr, and cron folder, since arrays cannot be passed into public methods other than the index() method as a single array with the current OC engine.

**Comment by TheCartpenter** [comment]
Will keep this one on the TODO list: https://forum.opencart.com/viewtopic.php?t=229559 .

**Comment by TheCartpenter** [comment]
Note: Instance variable in constructors involving a string type do need to be affected even though a string could be empty as per: https://github.com/opencart/opencart-3/pull/128#issuecomment-1285578320 .
